### DISCUSSION TITLE
2021-09-15 Weekly Planning

### DISCUSSION THREAD

**Discussion by mathemancer** [mathesar-foundation/mathesar#665] [Maintainer discussions]
### About
Each Tuesday, we start a discussion about Mathesar's progress. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

This discussion is abbreviated compared to the normal one.

**Comment by mathemancer** [comment]
## Questions

### Retrospective

- What went well last week?
- What could have gone better over the past week?
- Are there any improvements you'd like to suggest?


### Active Work

- Does your workload for this week look okay?
- The following milestones are due soon:
  - [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is due on Wednesday (the 15th).
  - [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34) is due next Thursday (the 23rd).

  Are there any dependencies or blockers that will prevent these being done on time?


### Discussion

Because this planning is somewhat more ad-hoc than normal, please be proactive about bringing up anything relevant for planning w.r.t. dependencies or blockers.

**Reply by mathemancer** [reply]
> What went well last week?

To be honest, last week was a bit rough.  Regardless, I got a PR submitted, and enjoyed reviewing @kgodey 's much-needed reorganization

> What could have gone better over the past week?

I realized too late that I was out-of-sync on what needed to be done for implementation of our custom money type.  This resulted in a lot of work going into something that isn't currently needed.

> Are there any improvements you'd like to suggest?

More vigilance from me w.r.t. what precisely is to be implemented.
 
> Does your workload for this week look okay?

I'm currently a bit behind due to the money type mix-up.  However, I think once I get that type sorted the others are much quicker/simpler to implement.

> The following milestones are due soon:
>   
> * [05. Editable Tables](https://github.com/centerofci/mathesar/milestone/4) is due on Wednesday (the 15th).
> * [06. 2021-09 Stability](https://github.com/centerofci/mathesar/milestone/34) is due next Thursday (the 23rd).
>   
>   Are there any dependencies or blockers that will prevent these being done on time?

I'm quite concerned about the progress of Milestone 06, but don't have any dependencies or blockers to mention.

> Because this planning is somewhat more ad-hoc than normal, please be proactive about bringing up anything relevant for planning w.r.t. dependencies or blockers.

Nothing comes to mind.  I'll update if it does.

**Reply by pavish** [reply]
> What went well last week?

  Did not go as well as I hoped it to. I got work done on different aspects but the progress on the milestones was static. Overall the week can be summed up to: PR for store reorganization is submitted, reviewed multiple frontend PRs (still more to go), interviews/hiring sync.

> What could have gone better over the past week?
> Are there any improvements you'd like to suggest?

  I discussed this with Kriti, and we agreed that it makes more sense to keep the scope of each issue/PR small enough be able to complete in a day or two.

Active Work
> Does your workload for this week look okay?
        The following milestones are due soon:
             05. Editable Tables is due on Wednesday (the 15th).
             06. 2021-09 Stability is due next Thursday (the 23rd).
>  Are there any dependencies or blockers that will prevent these being done on time?

The deadline for 5 is already crossed. I'm trying to wrap it up as quickly as possible. I expect it to be complete by Monday.

Milestone 6 has a lot of issues on the frontend, including setting up the integration test environment. I do not think we will be able to meet the deadline for it.

**Reply by ghislaineguerin** [reply]
> What went well last week?

Not a good week. Views took longer as I had to think through full flows rather than just isolated parts. Couldn't think about how users learn about tables vs. views yet.

> What could have gone better over the past week?

I think I should have discussed the issues before starting a bit more in-depth.

> Are there any improvements you'd like to suggest?

Not for now, I think the discussion should solve the problem described above.

> Does your workload for this week look okay?

Yes. I plan on opening a final review for milestone 7 last issue and start with UI revamp

> The following milestones are due soon:
> 05. Editable Tables is due on Wednesday (the 15th).
> 06. 2021-09 Stability is due next Thursday (the 23rd).
> Are there any dependencies or blockers that will prevent these from being done on time?

None
